integration-tests/src/test/java/com/datastax/oss/driver/api/core/ConnectIT.java

```diff
@@ -127,6 +133,36 @@ public void should_cleanup_on_lbp_init_failure() {
         .becomesTrue();
   }
 
+  /**
+   * Test for JAVA-2177. This ensures that even if the first attempted contact point is unreachable,
+   * its distance is set to LOCAL and reconnections are scheduled.
+   */
+  @Test
+  public void should_mark_unreachable_contact_points_as_local_and_schedule_reconnections() {
+    // Reject connections only on one node
+    BoundCluster boundCluster = simulacronRule.cluster();
+    boundCluster.node(0).rejectConnections(0, RejectScope.STOP);
+
+    try (CqlSession session = SessionUtils.newSession(simulacronRule)) {
+      Map<UUID, Node> nodes = session.getMetadata().getNodes();
+
+      // Before JAVA-2177, this would fail every other time because if the node was tried first for
+      // the initial connection, it was marked down and not passed to LBP.init(), and therefore
+      // stayed at distance IGNORED.
+      Node node0 = nodes.get(boundCluster.node(0).getHostId());
```

Comment:
Nice side-effect of JAVA-2165: finding a Simulacron node in the driver metadata is now straightforward.